### PR TITLE
Add Python scalar support to `dpt.where`

### PR DESCRIPTION
**Pull request by ndgrigorian**
This PR proposes a change to `dpt.where` to add support for Python scalars, aligning it with `clip` and elementwise functions.

As has been pointed out by the array API community, an common use-case for `dpt.where` is supporting Python scalars. If the decision to permit them in other elementwise functions is made, `where` will likely follow suit.

Regardless of the outcome of this decision, it's sensible to support them here.

As the first argument (`condition`) does not participate in the type promotion in `where`, if both arguments are scalars, the scalars are treated as the default types for the respective kind.

- [X] Have you provided a meaningful PR description?
- [X] Have you added a test, reproducer or referred to an issue with a reproducer?
- [X] Have you tested your changes locally for CPU and GPU devices?
- [X] Have you made sure that new changes do not introduce compiler warnings?
- [ ] Have you checked performance impact of proposed changes?
- [ ] If this PR is a work in progress, are you opening the PR as a draft?

### CODE REVIEW

File: dpctl/tensor/_search_functions.py [hunk 1]
```diff
@@ -81,36 +151,90 @@ def where(condition, x1, x2, /, *, order="K", out=None):
         raise TypeError(
             "Expecting dpctl.tensor.usm_ndarray type, " f"got {type(condition)}"
         )
-    if not isinstance(x1, dpt.usm_ndarray):
-        raise TypeError(
-            "Expecting dpctl.tensor.usm_ndarray type, " f"got {type(x1)}"
+    if order not in ["K", "C", "F", "A"]:
```
Comment (antonwolfy) [inline dpctl/tensor/_search_functions.py:154]: Do we need to update the docstring and to state scalars as possible type of input?

Comment (ndgrigorian) [reply dpctl/tensor/_search_functions.py:154]: Yes, good point

**Review by antonwolfy — APPROVED** [review]
I checked the PR in dpnp tests and all tests are passed on GPU and CPU devices.

**Comment by ndgrigorian** [comment]
CI failure is unrelated to this PR, I will go ahead and merge it

**Pull request merged**